app/src/main/java/com/example/proptit_protify/search/SearchFragment.kt

```diff
@@ -5,56 +5,50 @@ import androidx.fragment.app.Fragment
 import android.view.View
 import android.view.ViewGroup
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
 import com.example.proptit_protify.R
+import com.example.proptit_protify.databinding.FragmentSearchBinding
+import com.example.proptit_protify.service.MediaPlaybackService
 
-// TODO: Rename parameter arguments, choose names that match
-// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
-private const val ARG_PARAM1 = "param1"
-private const val ARG_PARAM2 = "param2"
-
-/**
- * A simple [Fragment] subclass.
- * Use the [SearchFragment.newInstance] factory method to
- * create an instance of this fragment.
- */
 class SearchFragment : Fragment() {
-    // TODO: Rename and change types of parameters
-    private var param1: String? = null
-    private var param2: String? = null
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        arguments?.let {
-            param1 = it.getString(ARG_PARAM1)
-            param2 = it.getString(ARG_PARAM2)
-        }
-    }
-
+    private var _binding: FragmentSearchBinding? = null
+    private val binding get() = _binding!!
+    private val searchViewModel: SearchViewModel by viewModels()
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_search, container, false)
+    ): View {
+        _binding = FragmentSearchBinding.inflate(inflater,container,false)
+        return binding.root
     }
-
-    companion object {
-        /**
-         * Use this factory method to create a new instance of
-         * this fragment using the provided parameters.
-         *
-         * @param param1 Parameter 1.
-         * @param param2 Parameter 2.
-         * @return A new instance of fragment SearchFragment.
-         */
-        // TODO: Rename and change types and number of parameters
-        @JvmStatic
-        fun newInstance(param1: String, param2: String) =
-            SearchFragment().apply {
-                arguments = Bundle().apply {
-                    putString(ARG_PARAM1, param1)
-                    putString(ARG_PARAM2, param2)
-                }
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val searchAdapter = SearchAdapter(
+            onSongClick = {data ->
+                this.navigateToNowPlayingScreen(data.id)
             }
+        )
+        //Dùng tạm nút Cancel làm nút Search
+        binding.cancel.setOnClickListener{
+            searchViewModel.search(binding.searchBar.text.toString())
+        }
+        binding.listSong.adapter = searchAdapter
+        binding.listSong.layoutManager = LinearLayoutManager(binding.root.context)
```

Comment:
orientation is not provided here, which may cause some strange ui on debug. considering remove this and use xml attribute `app:layoutManager`

Reply:
Em đã xóa dòng đó đi và chuyển về set chiều ở file fragment_search.xml ạ